package/MDAnalysis/auxiliary/base.py

```diff
@@ -0,0 +1,659 @@
+            self._read_next_step()
+            self._dt = self.time - self._initial_time
+            self.go_to_first_step()
+
+    def __len__(self):
+        """ Number of steps in auxiliary data. """
+        return self.n_steps
+
+    def next(self):
+        """ Move to next step of auxiliary data. """
+        self._read_next_step()
+        return {'time': self.time, 'data': self.step_data}
+
+
+    def __next__(self):
+        """ Move to next step of auxiliary data. """
+        return self.next()
+
+    def __iter__(self):
+        """ Iterate over all auxiliary steps. """
+        self._restart()
+        return self
+
+    def _restart(self):
+        """ Reset back to start; calling next() should read first step. """
+        # Overwrite as appropriate
+        self.step = -1
+                
+    def go_to_first_step(self):
+        """ Return to and read first step. """
+        # Overwrite as appropriate
+        #  could also use go_to_step(0)
+        self._restart()
+        self._read_next_step()
+                
+    def _read_next_step(self):
+        """ Move to next step and update data. """
+        # Define in each auxiliary reader
+        raise NotImplementedError(
+            "BUG: Override _read_next_timestep() in auxiliary reader!")
+
+    def read_ts(self, ts):
+        """ Read auxiliary data corresponding to the trajectory timestep *ts*.
+
+        Read the auxiliary steps 'assigned' to *ts* (the steps that are within
+        ``ts.dt/2`` of of the trajectory timestep/frame - ie. closer to *ts*
+        than either the preceeding or following frame). Then calculate a 
+        'representative value' for the timestep from the data in each of these 
+        auxiliary steps.
+
+        If ``auxname`` is set, the representative value will also be added to the
+        timestep as ``ts.aux.auxname``.
+
+        Parameters
+        ----------
+        ts : :class:`~MDAnalysis.coordinates.base.Timestep` object
+            The trajectory timestep for which corresponding auxiliary data is
+            to be read.
+
+        Returns
+        -------
+        :class:`~MDAnalysis.coordinates.base.Timestep`
+            If ``auxname`` is set, the representative auxiliary value in ``ts.aux`` 
+            will be updated appropriately.
+
+        Note
+        ----
+        The auxiliary reader will end up positioned at the last step assigned
+        to the trajectory frame or, if the frame includes no auxiliary steps,
+        (as when auxiliary data are less frequent), the most recent auxiliary 
+        step before the frame.
+        """
+        # Make sure our auxiliary step starts at the right point (just before
+        # the frame being read): the current step should be assigned to a 
+        # previous frame, and the next step to either the frame being read or a 
+        # following frame. Move to right position if not.
+        if not (self.step_to_frame(self.step, ts) < ts.frame
```

Comment:
This is doing a lot of file-searching and skipping in the file here. Sure we need all of this? Can't this be done in memory?

I'm asking because my home-folder where research data is stored is a network attached storage. They don't really like when I request searching that often.